pmd-java/src/test/java/net/sourceforge/pmd/lang/java/rule/bestpractices/CheckResultSetTest.java

```diff
@@ -0,0 +1,13 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.rule.bestpractices;
+
+public class CheckResultSetTest extends BestPracticesRulesTest {
+
+    @Override
+    public void setUp() {
+        addRule(RULESET, "CheckResultSet");
```

Comment:
Some implementation thoughts:

We could use convention the existing naming conventions to infer the name of the rule and the category reference.
* The rule name can be determined from the class name: `getClass().getSimpleName().replaceFirst("Test$", "")`
* The RULESET reference is also easy to infer from our package naming conventions, eg by matching the package name with the following regex: `.*lang\.(\w+?)\.rule\.(\w+)$`.

This would obviate the need for 
* having a base class for each category
* implementing anything in concrete rule test classes

If we migrate tests of all rulesets to that new format (single class per rule), then I think we could use another abstract class than SimpleAggregatorTst (since we don't aggregate anything anymore), e.g. ConventionalSingleRuleTest (internal api I guess, since it relies on our naming conventions), that would contain the implementation

Sidenote, in case we start using Kotlin to test rules, we could even avoid having a file for each test and just declare a bunch of classes in a single file (per category), e.g.:
```kotlin
class CheckResultSetTest : ConventionalSingleRuleTest()
class ConstantsInInterfaceTest : ConventionalSingleRuleTest()
...
```
which can be run individually

Reply:
Yes, we can use convention for the rule name. I think, I'll change that, it should work.
Using convention for the ruleset should work too, and I like the idea to have a separate base test class for our conventional based rule tests...
I'll try to update https://pmd.github.io/pmd-6.6.0/pmd_userdocs_extending_testing.html , too.

About naming of SimpleAggregatorTst: That's argueable - we still aggregate multiple tests into one test class :smile: But we could think of a better name for PMD 7 - maybe BaseRuleTest? It would be part of the public API to support unit testing custom rules.